src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
```

Comment:
Note that because these tests are running on xunit's SynchronizationContext, and because BackgroundWorker will post back its completion event to run on the original SynchronizationContext, care will need to be taken with such blocking.  Either you'd want to make the test be async and await something, or if you explicitly blocked you'd want to ensure you were on a null SynchronizationContext for the body of the method, as you did in other tests.